README.md
# bible.js

bible.js is a JavaScript package for both Node and the browser (via Bower) for handling Bible references. It can:

  * parse textual references into a standard, index based format
  * convert a textual reference to an absolute verse number
  * convert an absolute verse number to its textual reference
  * handle chapter references
  * handle ranges (e.g. John 3:1-16) (WIP)

# Installation

**node**

    npm install --save biblejs

# Usage

The `biblejs` module exposes three top level exports:

  const Bible = require('biblejs');
  let Reference = Bible.Reference // A class representing a single verse or chapter
  let Range = Bible.Range // A class representing a range of verses
  let Books = Bible.Books // A reference array containing data on the names and number of chapters/verses for each book in the Bible


## Reference

A `Reference` instance represents a reference to a specific Bible verse, or an entire chapter.

    var ref = new Reference("John 3:16");
    // returns a Reference instance for "John 3:16"

### Instance Methods

### isChapter()
Returns true if the reference is to an entire chapter rather than a specific verse

    var chapter = new Reference("John 1");
    var verse = new Reference("John 1:1");

    chapter.isChapter() // true
    verse.isChapter() // false


### startOf('book|chapter')
Modifies the reference in place (doesn't create a new one!), moving the reference to the start of the book or chapter.

    var ref = new Reference("John 3:16");

    ref.startOf('chapter').toString() // "John 3:1"
    ref.startOf('book').toString() // "John 1:1"


### clone()
Create a new Reference object of the same verse. Useful if you want to perform in-place mutations to the Reference (i.e. `.startOf()`) but want to retain the original reference as well.

### toString()
Returns the textual representation of the reference. Right now, this only supports full book names. Future versions will likely include a format argument to determine the format of the returned string (PR's welcome!).

    Reference.fromVerseId(1).toString() // "Genesis 1:1"

### toVerseId()
Returns the verse id (i.e. the number of the verse if you started from Genesis 1:1 and counted up to it).

    var ref = new Reference("Genesis 1:1");
    ref.toVerseId() // 1

### #toChapterId()
Returns the chapter id (i.e. the number of the chapter if you started from Genesis 1 and counted up to it).

    var ref = new Reference("Genesis 7");
    ref.toChapterId() // 7


### #toBookId()
Returns the book id (i.e. the number of the book if you started from Genesis and counted up to it).

    var ref = new Reference("Exodus");
    ref.toBookId() // 2

### Static Methods
The static methods found on the Reference class are useful for creating References from something other than a string (i.e. a verse or chapter id), as well as some utility methods for performing reference math.

### Reference.bookIdFromName(name)
Returns the book id given a book name. Handles most common book abbreviations.

    Reference.bookIdFromName("Genesis") // 1
    Reference.bookIdFromName("Gen") // 1
    Reference.bookIdFromName("Exo") // 2

#### `Reference.bookNameFromId(id)`
Get the name of the book of the Bible that matches the given id:

    Reference.bookNameFromId(1) // "Genesis"

#### Reference.fromChapterId(chapterId) {
Parse a chapter id into a reference. A chapter id is the 1-indexed absolute number of the chapter.

    Reference.fromChapterId(1) // Genesis 1

#### Reference.fromVerseId(verseId) {
Parse a verse id into a reference. A verse id is 1-indexed absolute number of the verse (i.e. Genesis 2:1's verse id is 32, because Genesis 1 has 31 verses).

    Reference.fromVerseId(1) // Genesis 1:1

#### Reference.versesInBookId(bookId) {
Get the number of verses in the given book id

    // How many verses are in all of Genesis?
    Reference.versesInBookId(1) // 1533

#### Reference.versesInChapterId(chapterId) {
Get the number of verses in the given chapter id

    // How many verses are in Genesis 1?
    Reference.versesInChapterId(1) // 31

#### Reference.chaptersInBookId(bookId) {
Get the number of chapters in the given book id

    // How many chapters are in Genesis?
    Reference.chaptersInBookId(1) // 50

#### Reference.versesUpToBookId(bookId) {
Get the number of verses up to the start of the given book id

    // How many verses are there in all the books before Exodus?
    Reference.versesUpToBookId(2) // 1533

#### Reference.versesUpToChapterId(chapterId) {
Get the number of verses up to the start of the given chapter id

    // How many verses are there in all the books before Exodus 2?
    Reference.versesUpToChapterId(51) // 1555

#### Reference.chaptersUpToBookId(bookId) {
Get the number of chapters up to the start of the given book id

    // How many chapters are there in all the books before Exodus?
    Reference.chaptersUpToBookId(2) // 50

### Reference.bookNameFromId(id)
Returns the full book name given a book id.

    Reference.bookNameFromId(1) // "Genesis"
    Reference.bookNameFromId(2) // "Exodus"


## Range
A `Range` instance represents a reference to a range of Bible verses. Internally, this is represented as to Reference objects that track the start and end of the range of verses.

    var start = new Reference('Genesis 1:1');
    var start = new Reference('Genesis 2:1');
    var range = new Range(start, end);
    // returns a Range instance that represents Genesis 1:1-2:1

### Instance Methods

#### distance()
Calculates the "distance" the Range covers. Returns an object that includes the number of verses, chapters, and books in the Range:

    var start = new Reference('Genesis 1:1');
    var start = new Reference('Exodus 1:1');
    var range = new Range(start, end);
    range.distance(); // { verses: 1533, chapters: 50, books: 1 }

A book or chapter is consided "in" the range if it's entire contents are. So Genesis 1:1 - Exodus 2:1 only has 1 book in the range, but has 51 chapters in the range (since all of Exodus is *not* in the range, but all of Exodus 1 *is* in the range).


## Books
This is a reference data array that includes the common names and their variants and abbreviations of each book of the Bible, along with the number of chapters in each book, and the number of verses in each chapter.


CHANGELOG.md
# 1.0.0
  * rewrite in vanilla JS
  * remove bower.json
  * add test coverage

# 0.0.8
  * added `CHANGELOG` and `dist/bible.js` to release task commit list
  * 0.0.7 forgot to include the updated built file for bower, re-releasing with it now

# 0.0.7
  * added `Reference.bookNameFromId`
  * fixed splitting of book names


test/test.js
'use strict';

var expect = require('chai').expect;
var Reference = require('../lib/index').Reference;

var validString = 'John 3:16'

describe('Reference', function() {
	describe('new Reference(referenceString)', function() {
		it('should create a reference object from a valid string construction', function() {
	        var ref = new Reference('John 3:16');
	        expect(ref).to.be.an('object');
	    });

	    it('should throw an error from invalid type passed to constructor', function() {
	        var refFn = function () {return new Reference(123)};
	        expect(refFn).to.throw(Error);
	    });
	});

	describe('Reference.fromVerseId(id)', function() {
		it('should create a reference object from a valid verse number', function() {
	        var ref = Reference.fromVerseId(4);
	        expect(ref.toString()).to.equal('Genesis 1:4');
	    });

	    it('should throw an error from invalid number passed to constructor', function() {
	        var refFn = function () {return Reference.fromVerseId(-4);};
	        expect(refFn).to.throw(Error);
	    });
	});

	describe('Reference.fromChapterId(id)', function() {
		it('should create a reference object from a valid chapter number', function() {
	        var ref = Reference.fromChapterId(4);
	        expect(ref.toString()).to.equal('Genesis 4');
	    });

	    it('should throw an error from invalid number passed to constructor', function() {
	        var refFn = function () {return Reference.fromChapterId(-4);};
	        expect(refFn).to.throw(Error);
	    });
	});

	describe('#.isChapter()', function() {
		it('should return true if the reference does not contain a verse', function() {
	        var ref = new Reference("John 1");
	        expect(ref.isChapter()).to.equal(true);
	    });

	    it('should return false if the reference contains a verse', function() {
	        var ref = new Reference("John 1:6");
	        expect(ref.isChapter()).to.equal(false);
	    });
	});

	describe('#startOf(\'book|chapter\')', function() {
		it('should modify the reference to refer to the start of the chapter if \'chapter\' is passed', function() {
	        var ref = new Reference('John 3:16');
	        expect(ref.startOf('chapter').toString()).to.equal('John 3:1');
	    });

	    it('should modify the reference to refer to the start of the book if \'book\' is passed', function() {
	        var ref = new Reference('John 3:16');
	        expect(ref.startOf('book').toString()).to.equal('John 1:1');
	    });
	});

	describe('#clone()', function() {
		it('should return a new Reference object to the same reference', function() {
	        var ref = new Reference('John 3:16');
	        var newRef = ref.clone();
	        newRef.startOf('book');
	        expect(ref.toString()).to.equal('John 3:16');
	    });
	});
	
	describe('#toString()', function() {
		it('should return the textual representation of the Reference', function() {
	        var ref = new Reference('John 3:16');
	        expect(ref.toString()).to.equal('John 3:16');
	    });
	});

	describe('#toVerseId()', function() {
		it('should return the verse id', function() {
	        var ref = new Reference('Genesis 1:16');
	        expect(ref.toVerseId()).to.equal(16);
	    });

	    it('should throw an error if the Reference has no verse', function() {
	        var refFn = function () {return new Reference('Genesis 1').toVerseId()};
	        expect(refFn).to.throw(Error);
	    });
	});

	describe('#toChapterId()', function() {
		it('should return the chapter id', function() {
	        var ref = new Reference('Exodus 6:16');
	        expect(ref.toChapterId()).to.equal(56);
	    });
	});

	describe('#toBookId()', function() {
		it('should return the book id', function() {
	        var ref = new Reference('Exodus 6:16');
	        expect(ref.toBookId()).to.equal(2);
	    });
	});

	describe('Reference.bookIdFromName(name)', function() {
		it('should return the book id given a book name', function() {
	        expect(Reference.bookIdFromName('Genesis')).to.equal(1);
	    });

	    it('should throw an error given an invalid book name', function() {
	    	var refFn = function () {return Reference.bookIdFromName('Hezekiah')};
	        expect(refFn).to.throw(Error);
	    });
	});

	describe('Reference.bookNameFromId(id)', function() {
		it('should return the book name given a book id', function() {
	        expect(Reference.bookNameFromId(1)).to.equal('Genesis');
	    });

	    it('should throw an error given an invalid book id', function() {
	        var refFn = function () {return Reference.bookNameFromId(-1)};
	        expect(refFn).to.throw(Error);
	    });
	});

	describe('Reference.versesInBookId(id)', function() {
		it('should return the number of verses in the given book id', function() {
	        expect(Reference.versesInBookId(1)).to.equal(1533);
	    });

	    it('should throw an error given an invalid book id', function() {
	        var refFn = function () {return Reference.versesInBookId(-1)};
	        expect(refFn).to.throw(Error);
	    });
	});

	describe('Reference.versesInChapterId(id)', function() {
		it('should return the number of verses in the given chapter id', function() {
	        expect(Reference.versesInChapterId(1)).to.equal(31);
	    });

	    it('should throw an error given an invalid chapter id', function() {
	        var refFn = function () {return Reference.versesInChapterId(0)};
	        expect(refFn).to.throw(Error);
	    });
	});

	describe('Reference.chaptersInBookId(id)', function() {
		it('should return the number of chapters in the given book id', function() {
	        expect(Reference.chaptersInBookId(1)).to.equal(50);
	    });

	    it('should throw an error given an invalid book id', function() {
	        var refFn = function () {return Reference.chaptersInBookId(67)};
	        expect(refFn).to.throw(Error);
	    });
	});

	describe('Reference.versesUpToBookId(id)', function() {
		it('should return the number of verses in all the books prior to the given book id', function() {
	        expect(Reference.versesUpToBookId(3)).to.equal(2746);
	    });

	    it('should throw an error given an invalid book id', function() {
	        var refFn = function () {return Reference.versesUpToBookId(67)};
	        expect(refFn).to.throw(Error);
	    });
	});

	describe('Reference.versesUpToChapterId(id)', function() {
		it('should return the number of verses in all the books prior to the given chapter id', function() {
	        expect(Reference.versesUpToChapterId(2)).to.equal(31);
	    });

	    it('should throw an error given an invalid chapter id', function() {
	        var refFn = function () {return Reference.versesUpToChapterId(1190)};
	        expect(refFn).to.throw(Error);
	    });
	});

	describe('Reference.chaptersUpToBookId(id)', function() {
		it('should return the number of chapters in all the books prior to the given book id', function() {
	        expect(Reference.chaptersUpToBookId(3)).to.equal(90);
	    });

	    it('should throw an error given an invalid book id', function() {
	        var refFn = function () {return Reference.chaptersUpToBookId(67)};
	        expect(refFn).to.throw(Error);
	    });
	});

});


test/range.js
import test from 'ava';
import Reference from '../lib/reference';
import Range from '../lib/range';

test('distance() returns the distance between the References', (t) => {
  let start = new Reference('Genesis 1:1');
  let end = new Reference('Exodus 1:1');
  let range = new Range(start, end);
  let distance = range.distance();
  t.is(distance.books, 1);
  t.is(distance.chapters, 50);
  t.is(distance.verses, 1533);
});


test/reference.js
import test from 'ava';
import Reference from '../lib/reference';

test('isChapter() returns true for chapter references', (t) => {
  let ref = new Reference('Genesis 1');
  t.true(ref.isChapter());
});

test('isChapter() returns false for verse references', (t) => {
  let ref = new Reference('Genesis 1:7');
  t.false(ref.isChapter());
});

test('isChapter() returns false for the first verse in a chapter', (t) => {
  let ref = new Reference('Genesis 1:1');
  t.false(ref.isChapter());
});

test('startOf("book") moves ref to start of book', (t) => {
  let ref = new Reference('Genesis 2:3');
  ref.startOf('book');
  t.is(ref.toVerseId(), 1);
});

test('startOf("chapter") moves ref to start of chapter', (t) => {
  let ref = new Reference('Genesis 1:4');
  ref.startOf('chapter');
  t.is(ref.toVerseId(), 1);
});

test('clone() creates a copy of the ref', (t) => {
  let ref1 = new Reference('Genesis 1:4');
  let ref2 = ref1.clone();
  ref1.startOf('chapter');
  t.not(ref1, ref2);
  t.not(ref1.toVerseId(), ref2.toVerseId());
});

test('toString() returns the text reference', (t) => {
  let ref = Reference.fromVerseId(32);
  t.is(ref.toString(), 'Genesis 2:1');
});

test('toVerseId() returns the verse id', (t) => {
  let ref = new Reference('Genesis 1:1');
  t.is(ref.toVerseId(), 1);
  ref = new Reference('Exodus 1:1');
  t.is(ref.toVerseId(), 1534);
});

test('toChapterId() returns the chapter id', (t) => {
  let ref = new Reference('Genesis 1:1');
  t.is(ref.toChapterId(), 1);
  ref = new Reference('Exodus 1:1');
  t.is(ref.toChapterId(), 51);
});

test('toBookId() returns the book id', (t) => {
  let ref = new Reference('Genesis 1:1');
  t.is(ref.toBookId(), 1);
  ref = new Reference('Exodus 1:1');
  t.is(ref.toBookId(), 2);
});

test('Reference.bookIdFromName() returns the book id', (t) => {
  t.is(Reference.bookIdFromName('Exodus'), 2);
});

test('Reference.bookNameFromId() returns the number of chapters in all books prior to the given book id', (t) => {
  t.is(Reference.bookNameFromId(2), 'Exodus');
});

test('Reference.fromChapterId() returns a reference instance for the chapter id', (t) => {
  t.is(Reference.fromChapterId(51).toString(), 'Exodus 1');
});

test('Reference.fromVerseId() returns a reference instance for the verse id', (t) => {
  t.is(Reference.fromVerseId(1534).toString(), 'Exodus 1:1');
});

test('Reference.versesInBookId() returns the total number of verses in that book', (t) => {
  t.is(Reference.versesInBookId(1), 1533); // Genesis
});

test('Reference.versesInChapterId() returns the total number of verses in that chapter', (t) => {
  t.is(Reference.versesInChapterId(51), 22); // Exodus 1
});

test('Reference.chaptersInBookId() returns the number of chapters in that book', (t) => {
  t.is(Reference.chaptersInBookId(1), 50); // Genesis
});

test('Reference.versesUpToBookId() returns the number of verses in all books prior to the given book id', (t) => {
  t.is(Reference.versesUpToBookId(3), 2746); // Leviticus
});

test('Reference.versesUpToChapterId() returns the number of verses in all chapters of all books prior to the given chapter id', (t) => {
  t.is(Reference.versesUpToChapterId(51), 1533); // Exodus 1
});

test('Reference.chaptersUpToBookId() returns the number of chapters in all books prior to the given book id', (t) => {
  t.is(Reference.chaptersUpToBookId(2), 50); // Exodus
});


gruntfile.js
module.exports = function(grunt) {
  grunt.registerTask("build", "Build the browserified bundle for browser use", ["browserify"]);
  grunt.registerTask("release", "Build, bump, and release", function(bumpArg) {
    if (bumpArg != null) {
      bumpArg = ":" + bumpArg;
    } else {
      bumpArg = "";
    }
    return grunt.task.run('build', "bump" + bumpArg, "publish");
  });
  grunt.initConfig({
    browserify: {
      all: {
        src: "lib/index.js",
        dest: "dist/bible.js"
      },
      options: {
        transform: ["coffeeify"],
        ignore: [require.resolve("coffee-script/register")],
        browserifyOptions: {
          extensions: [".coffee"]
        },
        bundleOptions: {
          standalone: "bible"
        }
      }
    },
    bump: {
      options: {
        files: ['package.json', 'bower.json'],
        commitFiles: ['package.json', 'bower.json', 'dist/bible.js', 'CHANGELOG.md'],
        pushTo: 'origin'
      }
    }
  });
  grunt.loadNpmTasks('grunt-browserify');
  grunt.loadNpmTasks('grunt-bump');
  return grunt.loadNpmTasks('grunt-npm');
};


lib/books.js
function generateOrdinalNameVariations(number, names) {
  var variations = [];
  var numerals;
  if (number === 1) {
    numerals = [ '1', 'I', 'First' ];
  } else if (number === 2) {
    numerals = [ '2' , 'II', 'Second' ];
  } else if (number === 3) {
    numerals = [ '3' , 'III', 'Third' ];
  }
  names.forEach(function(name) {
    numerals.forEach(function(numeral) {
      variations.push(numeral + name);
      variations.push(numeral + ' ' + name);
    });
  });
  return variations
}

module.exports = [
  {
    names: 'Genesis Ge Gen'.split(' '),
    verses: [31,25,24,26,32,22,24,22,29,32,32,20,18,24,21,16,27,33,38,18,34,24,20,67,34,35,46,22,35,43,55,32,20,31,29,43,36,30,23,23,57,38,34,34,28,34,31,22,33,26]
  },
  {
    names: 'Exodus Ex Exo'.split(' '),
    verses: [22,25,22,31,23,30,25,32,35,29,10,51,22,31,27,36,16,27,25,26,36,31,33,18,40,37,21,43,46,38,18,35,23,35,35,38,29,31,43,38]
  },
  {
    names: 'Leviticus Le Lev'.split(' '),
    verses: [17,16,17,35,19,30,38,36,24,20,47,8,59,57,33,34,16,30,37,27,24,33,44,23,55,46,34]
  },
  {
    names: 'Numbers Nu Num'.split(' '),
    verses: [54,34,51,49,31,27,89,26,23,36,35,16,33,45,41,50,13,32,22,29,35,41,30,25,18,65,23,31,40,16,54,42,56,29,34,13]
  },
  {
    names: 'Deuteronomy Dt Deut Deu De'.split(' '),
    verses: [46,37,29,49,33,25,26,20,29,22,32,32,18,29,23,22,20,22,21,20,23,30,25,22,19,19,26,68,29,20,30,52,29,12]
  },
  {
    names: 'Joshua Js Jos Jos Josh'.split(' '),
    verses: [18,24,17,24,15,27,26,35,27,43,23,24,33,15,63,10,18,28,51,9,45,34,16,33]
  },
  {
    names: 'Judges Jg Jud Jdg Ju Jdgs Judg'.split(' '),
    verses: [36,23,31,24,31,40,25,35,57,18,40,15,25,20,20,31,13,31,30,48,25]
  },
  {
    names: 'Ruth Ru Rut'.split(' '),
    verses: [22,23,18,22]
  },
  {
    names: generateOrdinalNameVariations(1, 'Samuel Sa Sam'.split(' ')),
    verses: [28,36,21,22,12,21,17,22,27,27,15,25,23,52,35,23,58,30,24,42,15,23,29,22,44,25,12,25,11,31,13]
  },
  {
    names: generateOrdinalNameVariations(2, 'Samuel Sa Sam'.split(' ')),
    verses: [27,32,39,12,25,23,29,18,13,19,27,31,39,33,37,23,29,33,43,26,22,51,39,25]
  },
  {
    names: generateOrdinalNameVariations(1, 'Kings Ki King Kin Kngs'.split(' ')),
    verses: [53,46,28,34,18,38,51,66,28,29,43,33,34,31,34,34,24,46,21,43,29,53]
  },
  {
    names: generateOrdinalNameVariations(2, 'Kings Ki King Kin Kngs'.split(' ')),
    verses: [18,25,27,44,27,33,20,29,37,36,21,21,25,29,38,20,41,37,37,21,26,20,37,20,30]
  },
  {
    names: generateOrdinalNameVariations(1, 'Chronicles Ch Chr Chron'.split(' ')),
    verses: [54,55,24,43,26,81,40,40,44,14,47,40,14,17,29,43,27,17,19,8,30,19,32,31,31,32,34,21,30]
  },
  {
    names: generateOrdinalNameVariations(2, 'Chronicles Ch Chr Chron'.split(' ')),
    verses: [17,18,17,22,14,42,22,18,31,19,23,16,22,15,19,14,19,34,11,37,20,12,21,27,28,23,9,27,36,27,21,33,25,33,27,23]
  },
  {
    names: 'Ezra Ez Ezr'.split(' '),
    verses: [11,70,13,24,17,22,28,36,15,44]
  },
  {
    names: 'Nehemiah Ne Neh Neh Ne'.split(' '),
    verses: [11,20,32,23,19,19,73,18,38,39,36,47,31]
  },
  {
    names: 'Esther Es Est Esth Ester'.split(' '),
    verses: [22,23,15,17,14,14,10,17,32,3]
  },
  {
    names: 'Job Jb Job'.split(' '),
    verses: [22,13,26,21,27,30,21,22,35,22,20,25,28,22,35,22,16,21,29,29,34,30,17,25,6,14,23,28,25,31,40,22,33,37,16,33,24,41,30,24,34,17]
  },
  {
    names: 'Psalm Ps Psa Pss Psalms'.split(' '),
    verses: [6,12,8,8,12,10,17,9,20,18,7,8,6,7,5,11,15,50,14,9,13,31,6,10,22,12,14,9,11,12,24,11,22,22,28,12,40,22,13,17,13,11,5,26,17,11,9,14,20,23,19,9,6,7,23,13,11,11,17,12,8,12,11,10,13,20,7,35,36,5,24,20,28,23,10,12,20,72,13,19,16,8,18,12,13,17,7,18,52,17,16,15,5,23,11,13,12,9,9,5,8,28,22,35,45,48,43,13,31,7,10,10,9,8,18,19,2,29,176,7,8,9,4,8,5,6,5,6,8,8,3,18,3,3,21,26,9,8,24,13,10,7,12,15,21,10,20,14,9,6]
  },
  {
    names: 'Proverbs Pr Prov Pro'.split(' '),
    verses: [33,22,35,27,23,35,27,36,18,32,31,28,25,35,33,33,28,24,29,30,31,29,35,34,28,28,27,28,27,33,31]
  },
  {
    names: 'Ecclesiastes Ec Ecc'.split(' '),
    verses: [18,26,22,16,20,12,29,17,18,20,10,14]
  },
  {
    names: ['Song of Solomon','SOS','Song of Songs', 'SongOfSongs'],
    verses: [17,17,11,16,16,13,13,14]
  },
  {
    names: 'Isaiah Isa'.split(' '),
    verses: [31,22,26,6,30,13,25,22,21,34,16,6,22,32,9,14,14,7,25,6,17,25,18,23,12,21,13,29,24,33,9,20,24,17,10,22,38,22,8,31,29,25,28,28,25,13,15,22,26,11,23,15,12,17,13,12,21,14,21,22,11,12,19,12,25,24]
  },
  {
    names: 'Jeremiah Je Jer'.split(' '),
    verses: [19,37,25,31,31,30,34,22,26,25,23,17,27,22,21,21,27,23,15,18,14,30,40,10,38,24,22,17,32,24,40,44,26,22,19,32,21,28,18,16,18,22,13,30,5,28,7,47,39,46,64,34]
  },
  {
    names: 'Lamentations La Lam Lament'.split(' '),
    verses: [22,22,66,22,22]
  },
  {
    names: 'Ezekiel Ek Ezek Eze'.split(' '),
    verses: [28,10,27,17,17,14,27,18,11,22,25,28,23,23,8,63,24,32,14,49,32,31,49,27,17,21,36,26,21,26,18,32,33,31,15,38,28,23,29,49,26,20,27,31,25,24,23,35]
  },
  {
    names: 'Daniel Da Dan Dl Dnl'.split(' '),
    verses: [21,49,30,37,31,28,28,27,27,21,45,13]
  },
  {
    names: 'Hosea Ho Hos'.split(' '),
    verses: [11,23,5,19,15,11,16,14,17,15,12,14,16,9]
  },
  {
    names: 'Joel Jl Joel Joe'.split(' '),
    verses: [20,32,21]
  },
  {
    names: 'Amos Am Amos Amo'.split(' '),
    verses: [15,16,15,13,27,14,17,14,15]
  },
  {
    names: 'Obadiah Ob Oba Obd Odbh'.split(' '),
    verses: [21]
  },
  {
    names: 'Jonah Jh Jon Jnh'.split(' '),
    verses: [17,10,10,11]
  },
  {
    names: 'Micah Mi Mic'.split(' '),
    verses: [16,13,12,13,15,16,20]
  },
  {
    names: 'Nahum Na Nah Na'.split(' '),
    verses: [15,13,19]
  },
  {
    names: 'Habakkuk Hb Hab Hk Habk'.split(' '),
    verses: [17,20,19]
  },
  {
    names: 'Zephaniah Zp Zep Zeph Ze'.split(' '),
    verses: [18,15,20]
  },
  {
    names: 'Haggai Ha Hag Hagg'.split(' '),
    verses: [15,23]
  },
  {
    names: 'Zechariah Zc Zech Zec'.split(' '),
    verses: [21,13,10,14,11,15,14,23,17,12,17,14,9,21]
  },
  {
    names: 'Malachi Ml Mal Mlc'.split(' '),
    verses: [14,17,18,6]
  },
  {
    names: 'Matthew Mt Matt Mat'.split(' '),
    verses: [25,23,17,25,48,34,29,34,38,42,30,50,58,36,39,28,27,35,30,34,46,46,39,51,46,75,66,20]
  },
  {
    names: 'Mark Mk Mrk'.split(' '),
    verses: [45,28,35,41,43,56,37,38,50,52,33,44,37,72,47,20]
  },
  {
    names: 'Luke Lk Luk Lu'.split(' '),
    verses: [80,52,38,44,39,49,50,56,62,42,54,59,35,35,32,31,37,43,48,47,38,71,56,53]
  },
  {
    names: 'John Jn Joh Jo'.split(' '),
    verses: [51,25,36,54,47,71,53,59,41,42,57,50,38,31,27,33,26,40,42,31,25]
  },
  {
    names: 'Acts Ac Act'.split(' '),
    verses: [26,47,26,37,42,15,60,40,43,48,30,25,52,28,41,40,34,28,41,38,40,30,35,27,27,32,44,31]
  },
  {
    names: 'Romans Ro Rom Rmn Rmns'.split(' '),
    verses: [32,29,31,25,21,23,25,39,33,21,36,21,14,23,33,27]
  },
  {
    names: generateOrdinalNameVariations(1, 'Corinthians Co Cor'.split(' ')),
    verses: [31,16,23,21,13,20,40,13,27,33,34,31,13,40,58,24]
  },
  {
    names: generateOrdinalNameVariations(2, 'Corinthians Co Cor'.split(' ')),
    verses: [24,17,18,18,21,18,16,24,15,18,33,21,14]
  },
  {
    names: 'Galatians Ga Gal Gltns'.split(' '),
    verses: [24,21,29,31,26,18]
  },
  {
    names: 'Ephesians Ep Eph Ephn'.split(' '),
    verses: [23,22,21,32,33,24]
  },
  {
    names: 'Philippians Phi Phil Phi'.split(' '),
    verses: [30,30,21,23]
  },
  {
    names: 'Colossians Co Col Colo Cln Clns'.split(' '),
    verses: [29,23,25,18]
  },
  {
    names: generateOrdinalNameVariations(1, 'Thessalonians Th Thess Thes'.split(' ')),
    verses: [10,20,13,18,28]
  },
  {
    names: generateOrdinalNameVariations(2, 'Thessalonians Th Thess Thes'.split(' ')),
    verses: [12,17,18]
  },
  {
    names: generateOrdinalNameVariations(1, 'Timothy Ti Tim'.split(' ')),
    verses: [20,15,16,16,25,21]
  },
  {
    names: generateOrdinalNameVariations(2, 'Timothy Ti Tim'.split(' ')),
    verses: [18,26,17,22]
  },
  {
    names: 'Titus Ti Tit Tt Ts'.split(' '),
    verses: [16,15,15]
  },
  {
    names: 'Philemon Pm Phile Philm Pm'.split(' '),
    verses: [25]
  },
  {
    names: 'Hebrews He Heb Hw'.split(' '),
    verses: [14,18,19,16,14,20,28,13,28,39,40,29,25]
  },
  {
    names: 'James Jm Jam Jas Ja'.split(' '),
    verses: [27,26,18,17,20]
  },
  {
    names: generateOrdinalNameVariations(1, 'Peter Pe Pet P'.split(' ')),
    verses: [25,25,22,19,14]
  },
  {
    names: generateOrdinalNameVariations(2, 'Peter Pe Pet P'.split(' ')),
    verses: [21,22,18]
  },
  {
    names: generateOrdinalNameVariations(1, 'John Joh Jo Jn J'.split(' ')),
    verses: [10,29,24,21,21]
  },
  {
    names: generateOrdinalNameVariations(2, 'John Joh Jo Jn J'.split(' ')),
    verses: [13]
  },
  {
    names: generateOrdinalNameVariations(3, 'John Joh Jo Jn J'.split(' ')),
    verses: [14]
  },
  {
    names: 'Jude Jude'.split(' '),
    verses: [25]
  },
  {
    names: 'Revelation Re Rev Rvltn'.split(' '),
    verses: [20,29,22,11,14,17,17,13,21,11,19,17,18,20,8,21,18,24,21,15,27,21]
  }
];
